File: mobile/packages/footprint_flutter/lib/src/onboarding-components/models/sandbox_outcome.dart
enum OverallOutcome {
  fail("fail"),
  pass("pass"),
  manualReview("manual_review"),
  useRulesOutcome("use_rules_outcome"),
  stepUp("step_up");

  final String value;

  const OverallOutcome(this.value);

  static OverallOutcome fromString(String value) {
    switch (value) {
      case "fail":
        return OverallOutcome.fail;
      case "pass":
        return OverallOutcome.pass;
      case "manual_review":
        return OverallOutcome.manualReview;
      case "use_rules_outcome":
        return OverallOutcome.useRulesOutcome;
      case "step_up":
        return OverallOutcome.stepUp;
      default:
        throw Exception("Invalid value $value");
    }
  }

  @override
  String toString() {
    return value;
  }
}

enum IdDocOutcome {
  pass("pass"),
  fail("fail"),
  real("real");

  final String value;

  const IdDocOutcome(this.value);

  static IdDocOutcome fromString(String value) {
    switch (value) {
      case "pass":
        return IdDocOutcome.pass;
      case "fail":
        return IdDocOutcome.fail;
      case "real":
        return IdDocOutcome.real;
      default:
        throw Exception("Invalid value $value");
    }
  }

  @override
  String toString() {
    return value;
  }
}

class SandboxOutcome {
  final OverallOutcome? overallOutcome;
  final IdDocOutcome? idDocOutcome;

  SandboxOutcome({
    this.overallOutcome,
    this.idDocOutcome,
  });

  factory SandboxOutcome.fromJson(Map<String, dynamic> json) {
    return SandboxOutcome(
      overallOutcome: OverallOutcome.fromString(json['overall_outcome']),
      idDocOutcome: IdDocOutcome.fromString(json['id_doc_outcome']),
    );
  }
}

File: mobile/NativeOnboardingComponents/shared/src/commonMain/kotlin/com/onefootprint/native_onboarding_components/models/SandboxOutcome.kt
package com.onefootprint.native_onboarding_components.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class OverallOutcome{
    @SerialName("pass") PASS {
        override fun toString(): String {
            return "pass"
        }
    },
    @SerialName("fail") FAIL {
        override fun toString(): String {
            return "fail"
        }
    },
    @SerialName("manual_review") MANUAL_REVIEW {
        override fun toString(): String {
            return "manual_review"
        }
    },
    @SerialName("use_rules_outcome") USE_RULES_OUTCOME {
        override fun toString(): String {
            return "use_rules_outcome"
        }
    },
    @SerialName("step_up") STEP_UP {
        override fun toString(): String {
            return "step_up"
        }
    }
}

@Serializable
enum class DocumentOutcome {
    @SerialName("pass") PASS {
        override fun toString(): String {
            return "pass"
        }
    },
    @SerialName("fail") FAIL {
        override fun toString(): String {
            return "fail"
        }
    },
    @SerialName("real") REAL {
        override fun toString(): String {
            return "real"
        }
    }
}

class SandboxOutcome(val overallOutcome: OverallOutcome, val documentOutcome: DocumentOutcome? = null)